tests/rptest/tests/schema_registry_test.py

```diff
@@ -1097,6 +1141,32 @@ def test_config(self):
         assert result_raw.json(
         )["message"] == f"Subject 'foo-key' not found.", f"{json.dumps(result_raw.json(), indent=1)}"
 
+    @cluster(num_nodes=3)
+    @parametrize(dataset_type=SchemaType.AVRO)
+    def test_normalize(self, dataset_type: SchemaType):
+        dataset = get_normalize_dataset(dataset_type)
+        self.logger.debug(f"testing with {dataset=}")
+
+        topic = create_topic_names(1)[0]
+
+        base_schema = json.dumps({
+            "schema": dataset.schema_base,
+            "schemaType": str(dataset.type)
+        })
+
+        self.logger.debug(
+            f"Register a schema against a subject - not normalized")
+        result_raw = self._post_subjects_subject_versions(
+            subject=f"{topic}-key", data=base_schema, normalize=False)
+        self.logger.debug(result_raw)
+        assert result_raw.status_code == requests.codes.ok
+        v1_id = result_raw.json()["id"]
+
+        self.logger.debug(f"Checking that the returned schema is canonical")
+        result_raw = self._get_schemas_ids_id(id=v1_id)
+        assert result_raw.status_code == requests.codes.ok
+        assert result_raw.json()['schema'] == dataset.schema_canonical
+
```

Comment:
If you rebase to dev, I've added a test with a TODO for the confluent SR client for normalization that might be nice to uncomment/implement now: https://github.com/redpanda-data/redpanda/pull/22328/files

Reply:
This will have to wait